File: lib/Models/RegionalModelInfoDTO.cpp


#include "RegionalModelInfoDTO.h"

using namespace Tiny;

RegionalModelInfoDTO::RegionalModelInfoDTO()
{
	currency = CurrencyType();
	regionId = long(0);
}

RegionalModelInfoDTO::RegionalModelInfoDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

RegionalModelInfoDTO::~RegionalModelInfoDTO()
{

}

void
RegionalModelInfoDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *currencyKey = "currency";

    if(object.has_key(currencyKey))
    {
        bourne::json value = object[currencyKey];




        CurrencyType* obj = &currency;
		obj->fromJson(value.dump());

    }

    const char *regionIdKey = "regionId";

    if(object.has_key(regionIdKey))
    {
        bourne::json value = object[regionIdKey];



        jsonToValue(&regionId, value, "long");


    }


}

bourne::json
RegionalModelInfoDTO::toJson()
{
    bourne::json object = bourne::json::object();






	object["currency"] = getCurrency().toJson();





    object["regionId"] = getRegionId();



    return object;

}

CurrencyType
RegionalModelInfoDTO::getCurrency()
{
	return currency;
}

void
RegionalModelInfoDTO::setCurrency(CurrencyType  currency)
{
	this->currency = currency;
}

long
RegionalModelInfoDTO::getRegionId()
{
	return regionId;
}

void
RegionalModelInfoDTO::setRegionId(long  regionId)
{
	this->regionId = regionId;
}

File: lib/app/Models/RegionalModelInfoDTO.php
<?php
/**
 * RegionalModelInfoDTO
 */
namespace app\Models;

/**
 * RegionalModelInfoDTO
 * @description Региональная информация.
 */
class RegionalModelInfoDTO {

    /** @var string $currency */
    public $currency = "";

    /** @var int $regionId Идентификатор региона, для которого выводится информация о предложениях модели (доставляемых в этот регион).  Информацию о регионе по идентификатору можно получить с помощью запроса [GET regions/{regionId}](../../reference/regions/searchRegionsById.md).*/
    public $regionId = 0;

}
